### PR TITLE
website: Update CLI and component docs

### PR DESCRIPTION
**Pull request by briancain**

### CODE REVIEW

File: website/content/partials/components/builder-pack.mdx [hunk 1]
```diff
@@ -67,6 +67,56 @@ The entrypoint binary is what provides extended functionality such as logs and e
...
+excluded. Git doesn’t list excluded directories for performance reasons,
+so any patterns on contained files have no effect, no matter where they
+are defined. Put a backslash ("\") in front of the first "!" for
+patterns that begin with a literal "!", for example, "\!important!.txt". 5. If the pattern ends with a slash, it is removed for the purpose of the
+following description, but it would only find a match with a directory.
+In other words, foo/ will match a directory foo and paths underneath it,
+but will not match a regular file or a symbolic link foo (this is
+consistent with the way how pathspec works in general in Git). 6. If the pattern does not contain a slash /, Git treats it as a shell glob
+pattern and checks for a match against the pathname relative to the
+location of the .gitignore file (relative to the toplevel of the work
```
Comment (catsby) [inline website/content/partials/components/builder-pack.mdx:88]:
```suggestion
location of the .gitignore file (relative to the top level of the work
```

File: website/content/partials/components/builder-pack.mdx [hunk 2]
```diff
@@ -67,6 +67,56 @@ The entrypoint binary is what provides extended functionality such as logs and e
 - Type: **bool**
 - **Optional**
 
+#### ignore
```
Comment (krantzinator) [inline website/content/partials/components/builder-pack.mdx:70]: This is kind of a mess, are we OK with that?

File: website/content/commands/server-upgrade.mdx [hunk 3]
```diff
@@ -27,13 +27,17 @@ Usage: `waypoint server upgrade [options]`
 
 - `-auto-approve` - Confirm server upgrade.
 - `-context-name=<string>` - Waypoint server context to upgrade.
-- `-platform=<string>` - Platform to upgrade the Waypoint server from.
+- `-platform=<string>` - Platform to upgrade the Waypoint server from, defaults to the platform stored in the context
```
Comment (krantzinator) [inline website/content/commands/server-upgrade.mdx:30]: nit: the original should say `from. Defaults to the platform stored in the context.

**Review by krantzinator — APPROVED** [review]

**Review by catsby — APPROVED** [review]
Minor suggestion and probably minor PR incoming

File: website/content/partials/components/builder-pack.mdx (hunk 2 above)
Comment (briancain) [reply website/content/partials/components/builder-pack.mdx:70]: I couldn't figure it out, so I copy pasted the text into the mdx file 😖 I'll open an issue to actually fix it, but for now I'd say good enough for getting the website looking nice.

**Pull request merged**